workflow-event-listener/src/main/java/com/netflix/conductor/contribs/listener/archive/ArchivingWorkflowListenerConfiguration.java

```diff
@@ -27,10 +27,14 @@ public class ArchivingWorkflowListenerConfiguration {
     @Bean
     public WorkflowStatusListener getWorkflowStatusListener(
             ExecutionDAOFacade executionDAOFacade, ArchivingWorkflowListenerProperties properties) {
-        if (properties.getTtlDuration().getSeconds() > 0) {
-            return new ArchivingWithTTLWorkflowStatusListener(executionDAOFacade, properties);
+        if (properties.getWorkflowArchivalType() == ArchivingWorkflowListenerProperties.ArchivalType.S3) {
+            return new ArchivingWorkflowToS3(executionDAOFacade, properties);
         } else {
-            return new ArchivingWorkflowStatusListener(executionDAOFacade);
+            if (properties.getTtlDuration().getSeconds() > 0) {
+                return new ArchivingWithTTLWorkflowStatusListener(executionDAOFacade, properties);
+            } else {
+                return new ArchivingWorkflowStatusListener(executionDAOFacade);
+            }
         }
```

Comment:
Can these be combined using else if instead of this?

Reply:
Done